### DISCUSSION TITLE
Codespaces don't seem to count Terminal activity when considering inactivity

### DISCUSSION THREAD

**Discussion by sheridanvk** [community/community#35428] [Codespaces]
Hi there - are there any plans to increase the scope of what counts as activity for keeping a codespace running? I often use them to carry out actions in the Terminal, but as far as I can tell, typing in the Terminal window doesn't seem to count as activity. I often find myself missing the message to keep the codespace from shutting down and losing my active session. 

I understand I can increase the timeout to make this happen less often, but it seems like a shame to keep it running longer than I actually need it - a 30min timeout would be totally fine for me if it actually took my Terminal activity into account.

Thank you!

**Comment by Kingfaya4201** [comment]
`
[Uploading Final advt. - UDIN (1).PDF…]()
`

**Comment by drien** [comment]
This would be super useful...working on things that take a long time to run/build/process is annoying with the existing idle timeout behavior, and led my team to find alternatives to Codespaces. I had a thought that a couple of tiny CLI utils for controlling this behavior would be neat...for example:

```bash
codespaces --caffeinate # keep alive indefinitely

codespaces --caffeinate --hours 3 # keep alive for 3 hours

make; codespaces --sleep-now # sleep once something long has finished
```

**Comment by EarthmanWeb** [comment]
> what counts as activity for keeping a codespace running

Expected behaviour - from:
https://docs.github.com/en/codespaces/getting-started/deep-dive#closing-or-stopping-your-codespace

> File changes from the editor and terminal output are counted as activity, so your codespace will not time out if terminal output is continuing.

So, it would seem yours is a feature request but mostly a bug report?  I have created a similar one here:
https://github.com/orgs/community/discussions/60407

**Comment by benonymus** [comment]
I am tailing a docker logs in the the terminal and after I close the codespace but still interact with the running app the codespace stops after the defined inactivity period.
Why is that if there is terminal output?
